astilectron.go

```diff
@@ -182,6 +185,48 @@ func (a *Astilectron) provision() error {
 	return a.provisioner.Provision(ctx, a.options.AppName, runtime.GOOS, runtime.GOARCH, *a.paths)
 }
 
+// function to create TCP/Unix socket connection
+func (a *Astilectron) establishConnection() (err error) {
+	/*
+	 * Switching between Unix-Socket and TCP-Socket
+	 * Windows will use TCP Socket
+	 * MAC and Linux will use Unix-Socket
+	 */
+	if runtime.GOOS == "windows" {
+		// Unfortunately communicating with Electron through stdin/stdout doesn't work on Windows so all communications
+		// will be done through TCP
+		if err = a.listenTCP(); err != nil {
+			return errors.Wrap(err, "TCP Socket listening failed")
+		}
+	} else {
+		if err = a.listenUnixSoc(); err != nil {
+			return errors.Wrap(err, "Unix Socket listening failed")
+		}
+	}
+	return nil
+}
+
+// Creates a unix socket
+func (a *Astilectron) listenUnixSoc() (err error) {
```

Comment:
Could you extract the duplicate code between `listenUnix` and `listenTCP` in a separate function? It should look something like:

```go
func (a *Astilectron) listenFunc(fn func() error) (err error) {
       // Log
	astilog.Debug("Listening...")

       // Custom
       if err = fn(); err != nil {
            err = errors.Wrap(err, "main: custom listen failed")
            return
       }

       // Check a connection has been accepted quickly enough
	var chanAccepted = make(chan bool)
	go a.watchNoAccept(a.options.AcceptTCPTimeout, chanAccepted)

	// Accept connections
	go a.acceptTCP(chanAccepted)
	return
}
```

That way `listenTCP` and `listenUnix` simply become something like:

```go
if err = a.listenFunc(fun() error {
    // Listen
	if a.listener, err = net.Listen("tcp", "127.0.0.1:"); err != nil {
		return errors.Wrap(err, "tcp net.Listen failed")
	}
}); err != nil {
    err = errors.Wrap(err, "main: listen func failed")
    return
}
```

And we avoid code duplication.

Reply:
reduced redundancy of code